src/Sarif/Core/StackFrame.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Reflection;
+using System.Text;
+using DotNetStackFrame = System.Diagnostics.StackFrame;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    /// A stack frame of a SARIF stack.
+    /// </summary>
+    public partial class StackFrame
+    {
+        internal const string IN = " in ";
+        internal const string AT = "   at ";
+        internal const string LINE = ":line";
+
+        /// <summary>
+        /// Creates a SARIF StackFrame instance from a .NET StackFrame instance
+        /// </summary>
+        /// <param name="stackTrace"></param>
+        /// <returns></returns>
+        public static StackFrame Create(System.Diagnostics.StackFrame dotNetStackFrame)
+        {
+            // This value is -1 if not present
+            string fileName = dotNetStackFrame.GetFileName();
+            int ilOffset = dotNetStackFrame.GetILOffset();
+            int nativeOffset = dotNetStackFrame.GetNativeOffset();
+            MethodBase methodBase = dotNetStackFrame.GetMethod();
+            Assembly assembly = methodBase?.DeclaringType.Assembly;
+            string fullyQualifiedName = CreateFullyQualifiedName(methodBase);
+
+            PhysicalLocation physicalLocation;
+            
+            if (fileName != null)
+            {
+                physicalLocation = new PhysicalLocation()
+                {
+                    Uri = new Uri(fileName),
+                    Region = new Region()
+                    {
+                        StartLine = dotNetStackFrame.GetFileLineNumber(),
+                        StartColumn = dotNetStackFrame.GetFileColumnNumber(),
+                        Offset = ilOffset
+                    }
+                };
+            }
+            else
+            {
+                physicalLocation = new PhysicalLocation()
+                {
+                    Region = new Region() { Offset = ilOffset }
+                };
+            }
+
+            StackFrame stackFrame = new StackFrame()
+            {
+                FullyQualifiedLogicalName = fullyQualifiedName,
```

Comment:
The schema includes a `LogicalLocationKey` property, but you don't populate it here -- and I don't even know how you'd figure it out!